sympy/physics/continuum_mechanics/beam.py

```diff
@@ -409,13 +419,13 @@ def join(self, beam, via="fixed"):
 
         if via == "fixed":
             new_beam = Beam(new_length, E, new_second_moment, x)
-            new_beam._composite_type = "fixed"
+            new_beam._joined_beam = True
             return new_beam
 
         if via == "hinge":
             new_beam = Beam(new_length, E, new_second_moment, x)
-            new_beam._composite_type = "hinge"
-            new_beam._hinge_position = self.length
```

Comment:
Are these not somehow needed in the current implementation? What are the consequences of deleting them?

Reply:
The new code is:
``` 
        if via == "hinge":
            new_beam = Beam(new_length, E, new_second_moment, x)
            new_beam._joined_beam = True
            new_beam.apply_hinge(self.length)
            return new_beam
```

This does the same as the old code, but in a different way. In the new way hinges can be placed anywhere in a beam instead of just the place where two beams are connected. The statement `new_beam._joined_beam = True` makes sure that it is known that the beam consists of two different parts that can have different E and I. This is necessary for slope and deflection calculations. The way of joining the two beams, fixed or by hinge, doesn't matter anymore because hinges are handled in a different way. 

The `new_beam.apply_hinge(self.length)` line places a hinge exacly at the place where the two beams are connected. and in the `apply_hinge` method it keeps track of all locations where hinges are placed. This replaceses the old line of `new_beam._hinge_position = self.length`. 

The main reason this is possible is because in the new way beams with and without hinges can be calculated in the same way. It doesn't matter anymore if there are hinges, so it is no more necessary to specify that the `_composite_type` is a hinge.